gmt-4.5.14/src/mgg/gmtinfo.1
.TH GMTINFO 1 "1 Nov 2015" "GMT 4.5.14" "Generic Mapping Tools"
.SH NAME
gmtinfo \- Get information about individual cruises
.SH SYNOPSIS
\fBgmtinfo\fP \fIleg-ids\fP
.SH DESCRIPTION
\fBgmtinfo\fP reports the minimum/maximum coordinates, year of the cruise, leg id, number of
records, the number of gravity, magnetics, and bathymetry points found, and the
start/stop dates for the leg.  All the information is written to standard output on
one line so that it is easy to use \fBawk\fP to add up numbers etc.
.TP
\fIleg-ids\fP
Can be one or more gmtleg-names, like c2610.  Use `cat list_of_legs` to pass all the legnames
in the file list_of_legs. 
.SH EXAMPLES
To obtain information about cruises c2603 and v2819, run
.br
.sp
\fBgmtinfo\fP  c2603 v2819 > info_file
.SH "SEE ALSO"
.IR GMT (1)
.SH REFERENCES
Wessel, P., and W. H. F. Smith, 2015, The Generic Mapping Tools (GMT) version
4.5.14 Technical Reference & Cookbook, SOEST/NOAA.
.br
Wessel, P., and W. H. F. Smith, 1998, New, Improved Version of Generic Mapping
Tools Released, EOS Trans., AGU, 79 (47), p. 579.
.br
Wessel, P., and W. H. F. Smith, 1995, New Version of the Generic Mapping Tools
Released, EOS Trans., AGU, 76 (33), p. 329.
.br
Wessel, P., and W. H. F. Smith, 1995, New Version of the Generic Mapping Tools
Released, http://www.agu.org/eos_elec/95154e.html, Copyright 1995 by the
American Geophysical Union.
.br
Wessel, P., and W. H. F. Smith, 1991, Free Software Helps Map and Display Data,
EOS Trans., AGU, 72 (41), p. 441.
.br


gmt-4.5.14/ChangeLog
-----------------------------------------------------------------------------------------
	$Id: ChangeLog 10331 2015-10-29 19:41:56Z pwessel $
-----------------------------------------------------------------------------------------

-----------------------------------------------------------------------------------------
2015-11-01	pwessel
	* GMT 4.5.14 finalized and released
-----------------------------------------------------------------------------------------
2015-10-29	pwessel
	* meca/psmeca.c:	Enforce rakes and strikes to be in -180/+180 range.
				Issue #799, for gmt4 only as not a problem in gmt5.

2015-07-28	pwessel
	* meca/psmeca.c:	Be able to read any value x and y coordinates

2015-06-24	pwessel
	* gmt_support.c:	Fix bug in jump near Dateline in gmt_inonout_sphpol_count,
				affecting inside/outside spherical cases, reported in message-2219

2015-03-18	pwessel
	* share/cpt/GMT_wysiwyg.cpt:	Fix typo in blue for high end [Issue #689].

2015-01-16	pwessel
	* gmt_plot.c:		Fix issue # 662 with graph arrows for negative scales.

2015-01-10	pwessel
	* gmt_plot.c:		Flipping the n/s sides array when lon and -y axes
				fixes issue #520

2015-01-01	pwessel
	* configure.ac:		Moved to 4.5.14 and back to svn release only.
-----------------------------------------------------------------------------------------
2015-01-01	pwessel
	* GMT 4.5.13 finalized and released [GSHHG 2.3.4 was released on Jan 1 as well]
-----------------------------------------------------------------------------------------
2014-12-28	pwessel
	* gmt_map.c:		Needed separate GMT_cartesian_overlap for non-geographic
				projections.  Fixes issue # 657.

2014-09-29	pwessel
	* gmt_plot.c:		Left a PSL variable undefined when grdimage plotted a grid
				with x = longitude and y = Cartesian.

2014-09-09	pwessel
	* grdfilter.c:		Fix y-shifts when -I sets smaller output.  Issue #616.
	* gmt_bcr.c:		Fix problem with points slightly outside causing an
				access violation; see issue #620.

2014-08-29	pwessel
	* sample1d.c:		Fix resampling for decreasing t values.  Issue #608.

2014-08-05	pwessel
	* gmt_map.c:		Could run into trouble clipping polygons for non-periodic
				maps.  Added more checks in GMT_wesn_clip.

2014-07-31	pwessel
	* gmt_map.c:		Rect clipping with -R<w/s/e/n>r could introduce stray lines
				when dataset has features about 180 degrees away from
				the user's area. See issue # 594.

2014-07-30	pwessel
	* configure.ac:		Set default GSHHG version to 2.3.2 (released August 1, 2014)
	* psxy[z].c:		Fixed wrap-around issue in fault lines.

2014-07-27	pwessel
	* gmtselect.c:		The -Z option needed to pass records with NaNs.

2014-07-19	pwessel
	* x2sys/x2sys_solve.c:	Fix problems that arise when a track has less crossovers
				than model parameters.
	* gmtmath.c:		Fix issue #123 with files and -C.

2014-07-01	pwessel
	* configure.ac:		Set default GSHHG version to 2.3.1 (released July 1, 2014)
	
2014-06-28	pwessel
	* mgd77/mgd77sniffer.c:	Needed to check that grid values used in decimate function
	 			also were within reasonable range.

2014-05-26	pwessel
	* gmt_map.c:		The map region clip path for -JE/e was hardwired to a full
				circle and did not care about -R setting.
				When -R...r and -Je/E was given the search for enclosing
				geographic boundaries failed when either the S or N pole
				was the projection's antipole.
2014-05-25	pwessel
	* grdmath.c:		Fix issue #566 bug in grdmath's KM2DEG operator.

2014-03-19	pwessel
	* gmtselect.c:		The -Fpolygon option might miss points because the io-machinery
				would sometimes set the min/max longitudes found incorrectly.

2014-03-14	pwessel
	* gmt_shore.c:		Give more error feedback if nc_open fails due to old library.

2014-03-01	pwessel
	* configure.ac:		Moved to 4.5.13 and back to svn release only.

-----------------------------------------------------------------------------------------
2014-03-01	pwessel
	* GMT 4.5.12 finalized and released [GSHHG 2.3.0 was released on Feb 1]
-----------------------------------------------------------------------------------------
2014-02-17	fwobbe
	* gmt_init.c etc.:	File locking -- if enabled via -DFLOCK -- is
				non-compulsory now.  This means if fcntl fails for some reason,
				GMT continues to access the file.  Hence, it is recommended to
				configure GMT with --enable-flock even if fcntl might be buggy
				on some ancient NFS shares.

2014-02-17	pwessel
	* gmt_map.c:		Size for memory allocation cast the product of nx * ny to
				size_t but overflow had already happen as calculation was
				done using ints, then cast to size_t. Must cast nx and ny
				separately.
2014-02-10	jluis
	* misc/gmt2kml.c:	Make sure the default style IDs are unique for each process.

2014-02-07	jluis
	* ps2raster.c:		Make sure that images in KML files do not carry a directory name

2014-02-03	pwessel
	* pslib.c:		Auto-widen paragraph width to at least fit widest word.

2014-02-01	pwessel
	* configure.ac:		Since GSHHG 2.3.0 was released today, make that the default

2014-01-26	pwessel
	* gmt_grdio.c:		Building the grid command string could exceed its length if
				the argument became exactly the max length of the array; then
				a final extra character \0 would be added and exceed array length.
				Thanks to Joachim Saul for pointing it out.

2014-01-22	pwessel
	* pscoast.c:		The processing of -I -N levels and pens could get out of sync
				so that the wrong pen was used for the specified feature.

2014-01-16	remko
	* pslib.c:		Take superscripts and subscripts into account when determining
				dimensions of text box.

2014-01-10	pwessel
	* grdmath.c:		Had to let SDIST return km instead of degrees since it cannot
				do degrees for geodesics.  By doing km it is now aligned with
				what PDIST and LDIST returns for geographic data, and matches
				what GMT5 does.  Some scripts had to be modified to handle the
				new behavior.  If you use SDIST you may wish to check your usage.
				Also fixed bug in SDIST in that spherical selected geodesics

gmt-4.5.14/src/mgg/gmtinfo.c
/*--------------------------------------------------------------------
 *	$Id: gmtinfo.c 10289 2014-12-28 21:17:06Z pwessel $
 *
 *    Copyright (c) 1991-2015 by P. Wessel and W. H. F. Smith
 *    See README file for copying and redistribution conditions.
 *--------------------------------------------------------------------*/
/*
 * gmtinfo prints out a string of information pertaining to each specified leg.
 * The parameters are: legname, year, no of records in file, no of gravity,
 * magnetics, and bathymetry points, area covered in lat/lon, and the d/m/yr
 * of the first and last records in the file(s).
 *
 * Author:	Paul Wessel
 * Date:	3-FEB-1988
 * Version:	3.2, 12-APR-1999 (DOS compatible)
 *
 */
#include "gmt.h"
#include "gmt_mgg.h"

int main (int argc, char **argv)
{
	int leg_year, n_records;
	int rec, n_grv, n_mag, n_top;
	int begin_day, begin_mo, begin_yr, end_day, end_mo, end_yr;
	int hour, minute, second, nfiles=0, i, verbose = FALSE;
	double xmin1, xmax1, xmin2, xmax2, ymin, ymax, xmin, xmax, lat, lon, old_x = 0.0;
	char agency[10], greenwich, gmtfile[100];
	FILE *fp = NULL;
	struct GMTMGG_REC record;
	struct GMTMGG_TIME *gmt = NULL;
	
	argc = (int)GMT_begin (argc, argv);

	gmtmggpath_init();
	
	for (i = 1; i < argc; i++) if (!strcmp(argv[i], "-v")) verbose = TRUE;
	
	for (i = 1; i < argc; i++) {	/* Loop over gmt files */
		if (argv[i][0] == '-') continue;
  		if (gmtmggpath_func (gmtfile, argv[i])) {
   			fprintf(stderr,"gmtinfo : Cannot find leg %s\n", argv[i]);
     			continue;
  		}
		if ((fp = fopen (gmtfile, "rb")) == NULL) {
			fprintf(stderr,"gmtinfo: Could not open file %s\n", gmtfile);
			continue;
		}
		nfiles++;
	
		if (verbose) fprintf (stderr, "Now reading leg %s\n", argv[i]);
		
		/* Read first record of file containing start-year, n_records and info */
	
		if (fread((void *)(&leg_year), (size_t)4, (size_t)1, fp) != 1) {
			fprintf(stderr,"gmtinfo: Error while reading first year\n");
			exit (EXIT_FAILURE);
		}
		if (fread((void *)(&n_records), (size_t)4, (size_t)1, fp) != 1) {
			fprintf(stderr,"gmtinfo: Error while reading no of records\n");
			exit (EXIT_FAILURE);
		}
		if (fread((void *)agency, (size_t)10, (size_t)1, fp) != 1) {
			fprintf(stderr,"gmtinfo: Error while reading info-header\n");
			exit (EXIT_FAILURE);
		}
	
		gmt = gmtmgg_init (leg_year);	/* Initialize gmt_structure */
	
		/* Start reading data from file */
	
		xmin1 = xmin2 = 360.0;
		xmax1 = xmax2 = -360.0;
		ymin = 180.0;
		ymax = -180.0;
		n_grv = n_mag = n_top = 0;
		greenwich = FALSE;
		for (rec = 0; rec < n_records; rec++) {
			if (fread((void *)(&record), (size_t)18, (size_t)1, fp) != 1) {
				fprintf(stderr,"gmtinfo: Error reading data record no %d\n",rec);
				exit (EXIT_FAILURE);
			}
			if (rec == 0)
				gmtmgg_date(record.time,&begin_yr,&begin_mo,&begin_day,&hour,&minute,&second,gmt);

			if (record.gmt[0] != GMTMGG_NODATA) n_grv++;
			if (record.gmt[1] != GMTMGG_NODATA) n_mag++;
			if (record.gmt[2] != GMTMGG_NODATA) n_top++;
			lon = record.lon * 0.000001;
			lat = record.lat * 0.000001;
			if (lon < 180.0) {
				xmin1 = MIN (lon, xmin1);
				xmax1 = MAX (lon, xmax1);
			}
			else {
				xmin2 = MIN (lon, xmin2);
				xmax2 = MAX (lon, xmax2);
			}
			ymin = MIN (lat, ymin);
			ymax = MAX (lat, ymax);
			if (rec > 0 && (fabs(old_x-lon) > 180.0))
				greenwich = TRUE;
			old_x = lon;
		}
		fclose (fp);
		
		gmtmgg_date (record.time,&end_yr,&end_mo,&end_day,&hour,&minute,&second,gmt);
		GMT_free ((void *)gmt);

		if (greenwich) {
			xmin = MAX(xmin1,xmin2);
			xmax = MIN(xmax1,xmax2);
		}
		else {
			xmin = MIN(xmin1,xmin2);
			xmax = MAX(xmax1,xmax2);
		}
		if (xmin > xmax) xmin -= 360.0;
		printf("Info: %s Year: %d N_recs: %d n_g: %d n_m: %d n_t: %d W: %.5f E: %.5f S: %.5f N: %.5f Begins: %d %d %d Ends: %d %d %d\n",
			agency, leg_year, n_records, n_grv, n_mag, n_top, xmin, xmax, ymin, ymax,
			begin_day, begin_mo, begin_yr, end_day, end_mo, end_yr);
	}
	
	if (nfiles == 0) {
		fprintf(stderr, "usage: gmtinfo leg(s)\n");
		exit (EXIT_FAILURE);
	}

	gmtmgg_end ();

	GMT_end (argc, argv);
	
	exit (EXIT_SUCCESS);
}
